src/workers/cdc/primary.ts

```diff
@@ -809,12 +817,59 @@ const onSquadPublicRequestChange = async (
   });
 };
 
+const setRestoreStreakCache = async (
+  con: DataSource,
+  streak: ChangeObject<UserStreak>,
+) => {
+  const { userId, currentStreak: previousStreak } = streak;
+  const today = new Date();
+  const shouldAllow = await shouldAllowRestore(con, streak);
+
+  if (!shouldAllow) {
+    return;
+  }
+
+  const key = generateStorageKey(StorageTopic.Streak, StorageKey.Reset, userId);
+  const now = today.getTime();
+  const nextDay = addDays(now, 1).setHours(0, 0, 0, 0);
+  const differenceInSeconds = (nextDay - now) * 1000;
+
+  await Promise.all([
+    setRedisObjectWithExpiry(key, previousStreak, differenceInSeconds),
+    con.getRepository(Alerts).update({ userId }, { showRecoverStreak: true }),
+  ]);
+};
+
+export const getRestoreStreakCache = async ({
+  userId,
+}: {
+  userId: User['id'];
+}): Promise<null | number> => {
+  const key = generateStorageKey(StorageTopic.Streak, StorageKey.Reset, userId);
+  const oldStreakLength = Number(await getRedisObject(key));
+  const userDoesntHaveOldStreak =
+    !oldStreakLength || !isNumber(oldStreakLength);
+
+  if (userDoesntHaveOldStreak) {
+    return null;
+  }
+
+  return oldStreakLength;
+};
+
 const onUserStreakChange = async (
   con: DataSource,
   logger: FastifyBaseLogger,
   data: ChangeMessage<UserStreak>,
 ) => {
   if (data.payload.op === 'u') {
+    if (
+      data.payload.after.currentStreak === 0 &&
+      data.payload.before.currentStreak > 2
```

Comment:
Can we move `2` to some constant, does the day gap one matches?

Reply:
I think this should be something else, let me update it.